src/System.Private.TypeLoader/src/Internal/TypeSystem/RuntimeNoMetadataType.cs

```diff
@@ -280,6 +283,69 @@ public override TypeDesc UnderlyingType
             }
         }
 
+        private void GetTypeNameHelper(out string name, out string nsName, out string assemblyName)
+        {
+            TypeReferenceHandle typeRefHandle;
+            QTypeDefinition qTypeDefinition;
+            MetadataReader reader;
+
+            RuntimeTypeHandle genericDefinitionHandle = GetTypeDefinition().GetRuntimeTypeHandle();
+            Debug.Assert(!genericDefinitionHandle.IsNull());
+
+            // NOTE: We're ignoring the containing type for compatiblity with SigFormat.cpp
```

Comment:
> I saw in other places that we also ignore the containing type name

It's one place: that place is formatting a display name for a type. Display name for a type can be e.g. `Namespace.Type+Nested[]`.

What you have here is reading type name and namespace as represented in the metadata. This is not supposed to do any formatting - just take the two strings from metadata and give them back. Adding a comment referencing formatting is just confusing. We're not trying to be compatible with anything here.